README.md

```diff
@@ -1,31 +1,120 @@
+</div>
+
+
+<br/>
+
+<p align="center">
+    <a href='https://img.shields.io/npm/v/necord'><img src="https://img.shields.io/npm/v/necord" alt="NPM Version" /></a>
+    <a href='https://img.shields.io/npm/l/necord'><img src="https://img.shields.io/npm/l/necord" alt="NPM License" /></a>
+    <a href='https://img.shields.io/npm/dm/necord'><img src="https://img.shields.io/npm/dm/necord" alt="NPM Downloads" /></a>
+    <a href='https://img.shields.io/github/last-commit/necordjs/necord'><img src="https://img.shields.io/github/last-commit/SocketSomeone/necord" alt="Last commit" /></a>
+</p>
+
+
+## About
+
+Transform your bot into a professional DJ with the power of the [Lavalink](https://lavalink.dev/) ecosystem. This package uses [lavalink-client](https://github.com/Tomato6966/lavalink-client) behind the scenes, providing a high-performance and efficient solution for managing audio streams on Discord. By leveraging Lavalink, your bot gains the ability to manage audio playback, queues, and real-time controls with minimal latency, transforming it into a fully capable and professional music system.
+
+## Installation
+
+**Node.js 18.0.0 or newer is required.**
+
+```bash
+$ npm i @necord/lavalink necord discord.js
+$ yarn add @necord/lavalink necord discord.js
+$ pnpm add @necord/lavalink necord discord.js
+```
+
+**Java 17.0.0 or newer is required.** \
+**Lavalink Server v4 or newer is required.** \
+[Lavalink Intallation Guide](https://lavalink.dev/configuration/)
 
 ## Usage
-1. Click the "Use this template" button to create a new repository from this template.
-2. Clone the new repository to your local machine.
-3. Run `npm install` to install dependencies.
-4. Replace name, description, and other fields in `package.json` with your own.
-5. Replace the contents of this file with your own.
-6. Replace name of package at github actions workflow file
-
-## Scripts
-* `npm run build` - Compile TypeScript to JavaScript.
-* `npm run lint` - Lint TypeScript files.
-* `npm run test` - Run tests.
-* `npm run publish:npm` - Publish package to npm.
-* `npm run publish:dev` - Publish package to npm with `dev` tag.
+
+Once the installation process is complete, we can import the `NecordLavalinkModule` into the root `AppModule`:
+
+```typescript
+import { NecordLavalinkModule } from '@necord/lavalink';
+import { Module } from '@nestjs/common';
+import { Client } from 'discord.js';
+import { AppUpdate } from './app.update';
+
+@Module({
+    imports: [
+        NecordLavalinkModule.forRootAsync({
+            useFactory: (client: Client) => ({
+                nodes: [
+                    {
+                        authorization: 'youshallnotpass',
+                        host: 'localhost',
+                        port: 2333,
+                        id: 'main_node'
+                    }
+                ],
+                sendToShard: (guildId, payload) => client.guilds.cache.get(guildId)?.shard?.send(payload),
```

Comment:
Can we process it by default? If it set not override

Reply:
Yeah, we can Inject `Client` in the `LavalinkManagerProvider` and handle it by default
```TypeScript
import { Provider } from '@nestjs/common';
import { LAVALINK_MODULE_OPTIONS } from '../necord-lavalink.module-definition';
import { GuildShardPayload, LavalinkManager } from 'lavalink-client';
import { ManagerOptions } from 'lavalink-client/dist/types/structures/Types/Manager';
import { Client } from 'discord.js';

export const LavalinkManagerProvider: Provider<LavalinkManager> = {
	provide: LavalinkManager,
	useFactory: (options: ManagerOptions, client: Client) =>
		new LavalinkManager({
			...options,
			sendToShard: (guildId: string, payload: GuildShardPayload) =>
				client.guilds.cache.get(guildId)?.shard?.send(payload)
		}),
	inject: [LAVALINK_MODULE_OPTIONS, Client]
};

```

Reply:
But removing this dependency we will have to handle the [client](https://github.com/necordjs/lavalink/pull/5/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R62-R65) configuration

Reply:
Yeah, but can we not remove it dependency? Just make it optional with default values which was already declared?

Reply:
I think removing this fields make it less flexible

Reply:
Then we can set these default values directly in the module providers right?
```TypeScript
import {
	ConfigurableModuleClass,
	LAVALINK_MODULE_OPTIONS
} from './necord-lavalink.module-definition';
import { Global, Logger, Module, OnApplicationShutdown, OnModuleInit } from '@nestjs/common';
import * as ProvidersMap from './providers';
import { DiscoveryModule } from '@nestjs/core';
import { DestroyReasons, GuildShardPayload, LavalinkManager } from 'lavalink-client';
import { Client } from 'discord.js';

const Providers = Object.values(ProvidersMap);

@Global()
@Module({
	imports: [DiscoveryModule],
	providers: [
		...Providers,
		{
			provide: LAVALINK_MODULE_OPTIONS,
			useFactory: (client: Client) => ({
				sendToShard: (guildId: string, payload: GuildShardPayload) =>
					client.guilds.cache.get(guildId)?.shard?.send(payload)
			}),
			inject: [Client]
		}
	],
	exports: [...Providers, LAVALINK_MODULE_OPTIONS]
})
export class NecordLavalinkModule
// ...
```

Reply:
Then we can set these default values directly in the module providers right?
```TypeScript
import {
	ConfigurableModuleClass,
	LAVALINK_MODULE_OPTIONS
} from './necord-lavalink.module-definition';
import { Global, Logger, Module, OnApplicationShutdown, OnModuleInit } from '@nestjs/common';
import * as ProvidersMap from './providers';
import { DiscoveryModule } from '@nestjs/core';
import { DestroyReasons, GuildShardPayload, LavalinkManager } from 'lavalink-client';
import { Client } from 'discord.js';

const Providers = Object.values(ProvidersMap);

@Global()
@Module({
	imports: [DiscoveryModule],
	providers: [
		...Providers,
		{
			provide: LAVALINK_MODULE_OPTIONS,
			useFactory: (client: Client) => ({
				sendToShard: (guildId: string, payload: GuildShardPayload) =>
					client.guilds.cache.get(guildId)?.shard?.send(payload)
			}),
			inject: [Client]
		}
	],
	exports: [...Providers, LAVALINK_MODULE_OPTIONS]
})
export class NecordLavalinkModule
// ...
```

Reply:
Implemented in: https://github.com/necordjs/lavalink/pull/5/commits/72618c20a47553b4a12c51b7fa868f8c8a680d72